### DISCUSSION TITLE
How to use client to receive the buffer

### DISCUSSION THREAD

**Discussion by Decodetalkers** [ebkalderon/tower-lsp#349] [Q&A] [answered]
I have notice the function of send_request. Can I use it to request the full buffer from the client?

**Comment by Decodetalkers** [comment, marked as answer]
And I have accomplished the hover, but my editor cannot send request to my lsp. Is the hover in the trait the lsp hover?

**Reply by ebkalderon** [reply]
Hello! The `Client::send_request` method is intended to send custom JSON-RPC 2.0 requests to the client which aren't defined by the LSP spec. AFAIK, there is no way to request the full buffer from the client in this manner. Instead, I believe the solution in this thread should do: https://github.com/ebkalderon/tower-lsp/discussions/346#discussioncomment-3031466
